packages/x-tree-view/src/themeAugmentation/components.d.ts

```diff
@@ -4,12 +4,12 @@ export interface TreeViewComponents<Theme = unknown> {
   MuiTreeItem?: {
     defaultProps?: ComponentsProps['MuiTreeItem'];
     styleOverrides?: ComponentsOverrides<Theme>['MuiTreeItem'];
-    variants?: ComponentsVariants['MuiTreeItem'];
+    variants?: ComponentsVariants<Theme>['MuiTreeItem'];
```

Comment:
I think we should bump it indeed.

By the way, one could say that bumping all our core peer deps to a quite recent version is a good idea in a major.
Because it reduces the risk that we will need to bump it in a minor to have access to some new exported utils or component.

Reply:
> By the way, one could say that bumping all our core peer deps to a quite recent version is a good idea in a major.

Let's discuss it after the holidays and if there are no objections - we can bump it for all the packages and after that merge this PR. 👍

Reply:
Thank you both 😄

Reply:
Hello @dhulme, we agreed and bumped the minimum `@mui/material` peer dependency: https://github.com/mui/mui-x/pull/11692.
Could you please update your PR by merging the latest `upstream/next` and updating the solution, because there were changes in components?
We can't do it on this PR, because it was raised from your fork's `next` branch, instead of a sub-branch and that one is protected. 🙈

Reply:
Thanks @LukasTy, I've updated the PR!

Reply:
Thank you for your help. 🙏